Jasmine/Common/ViewModels/CountDownTimer.swift

```diff
@@ -2,38 +2,40 @@ import Foundation
 
 /// A timer to handle game time
 class CountDownTimer {
+    typealias VMProtocol = (TimedViewModelProtocol & BaseViewModelProtocol)
 
-    /// Specifies the time remaining in the game.
-    private(set) var timeRemaining: TimeInterval
-
+    /// The ViewModel that contains this timer.
+    private var viewModel: VMProtocol?
     /// Specifies the total time allowed in the game.
     let totalTimeAllowed: TimeInterval
-
-    /// Retrieves time elapsed since game started
-    var timeElapsed: TimeInterval {
-        return totalTimeAllowed - timeRemaining
+    /// Specifies the time remaining in the game.
+    private(set) var timeRemaining: TimeInterval {
+        didSet {
+            viewModel?.timeDidUpdate(timeRemaining: timeRemaining, totalTime: totalTimeAllowed)
+        }
     }
 
     /// Initialize the total time allowed and sets timeRemaining to it
-    init(totalTimeAllowed: TimeInterval) {
+    init(totalTimeAllowed: TimeInterval, viewModel: VMProtocol?) {
         assert(totalTimeAllowed >= 0, "totalTimeAllowed must be positive")
+        self.viewModel = viewModel
         self.totalTimeAllowed = totalTimeAllowed
         timeRemaining = totalTimeAllowed
+        // Need to include this on init only
+        viewModel?.timeDidUpdate(timeRemaining: timeRemaining, totalTime: totalTimeAllowed)
     }
 
     /// Starts the countdown timer
-    func startTimer(timerInterval: TimeInterval, viewControllerDelegate: BaseGameViewControllerDelegate?) {
+    func startTimer(timerInterval: TimeInterval) {
         assert(timerInterval >= 0, "timerInterval must be positive")
 
-        viewControllerDelegate?.redisplay(timeRemaining: timeRemaining, outOf: totalTimeAllowed)
-        viewControllerDelegate?.notifyGameStatus(with: .inProgress)
+        viewModel?.gameStatus = .inProgress
```

Comment:
Seems like this class is too coupled. Like I can't really reason why should the timer handle the state of the game in VM? Shouldn't timer be dumb too?